lib/SILOptimizer/Transforms/StackPromotion.cpp

```diff
@@ -159,7 +159,7 @@ class StackPromoter {
     if (!PostDomTreeValid) {
       // The StackPromoter acts as a "graph" for which the post-dominator-tree
       // is calculated.
-      PostDomTree.recalculate(*this);
+      PostDomTree.recalculate(*F);
```

Comment:
I intentionally did not make this change in my patch because it's semantics-breaking: it causes the ordinary post-dominance tree to be built and therefore stops using the modified post-dominance order.  That's why I asked @eeckstein to look at it.